src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -563,6 +584,27 @@ private LocalSymbol GetDeclaredLocal(CSharpSyntaxNode declarationSyntax, SyntaxT
             return null;
         }
 
+        private LocalFunctionSymbol GetDeclaredLocalFunction(LocalFunctionStatementSyntax declarationSyntax, SyntaxToken declaredIdentifier)
+        {
+            return GetDeclaredLocalFunction(this.GetEnclosingBinder(GetAdjustedNodePosition(declarationSyntax)), declaredIdentifier);
+        }
+
+        private static LocalFunctionSymbol GetDeclaredLocalFunction(Binder enclosingBinder, SyntaxToken declaredIdentifier)
+        {
+            for (var binder = enclosingBinder ; binder != null; binder = binder.Next)
```

Comment:
This is not crucial, but since is a recurring pattern, this makes me think maybe an enumerator for binder chain would be nice.